Source/WebCore/css/StyleProperties.cpp

```diff
@@ -551,36 +512,40 @@ String StyleProperties::fontValue() const
 
     // Font stretch values can only be serialized in the font shorthand as keywords, since percentages are also valid font sizes.
     // If a font stretch percentage can be expressed as a keyword, then do that.
-    ASCIILiteral stretchPercentageAsKeyword;
-    bool stretchIsNormal = false;
+    std::optional<CSSValueID> stretchKeyword;
     if (auto stretchBase = getPropertyCSSValue(CSSPropertyFontStretch)) {
         auto stretch = downcast<CSSPrimitiveValue>(stretchBase.get());
-        std::optional<CSSValueID> keyword;
         if (!stretch->isPercentage())
-            keyword = stretch->valueID();
+            stretchKeyword = stretch->valueID();
         else {
-            keyword = fontStretchKeyword(stretch->doubleValue());
-            if (!keyword)
+            stretchKeyword = fontStretchKeyword(stretch->doubleValue());
+            if (!stretchKeyword)
                 return emptyString();
-            stretchPercentageAsKeyword = nameLiteral(*keyword);
         }
-        stretchIsNormal = keyword == CSSValueNormal;
     }
 
-    // This code no longer uses commonValue, for now we define it so we can use appendFontLonghandValueIfExplicit.
-    String commonValue;
+    auto appendValueIfExplicit = [this](CSSPropertyID property, StringBuilder& result) {
+        int foundPropertyIndex = findPropertyIndex(property);
+        if (foundPropertyIndex == -1)
+            return; // All longhands must have at least implicit values if "font" is specified.
+
+        if (propertyAt(foundPropertyIndex).isImplicit())
+            return;
```

Comment:
Later we need to change this to be based on the value of the properties, not on `isImplicit`.